ddl/placement_policy.go

```diff
@@ -232,3 +241,53 @@ func onAlterPlacementPolicy(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	job.FinishDBJob(model.JobStateDone, model.StatePublic, ver, nil)
 	return ver, nil
 }
+
+func checkPlacementPolicyNotInUse(d *ddlCtx, t *meta.Meta, policy *model.PolicyInfo) error {
+	currVer, err := t.GetSchemaVersion()
+	if err != nil {
+		return err
+	}
+	is := d.infoCache.GetLatest()
+	if is.SchemaMetaVersion() == currVer {
+		return checkPlacementPolicyNotInUseFromInfoSchema(is, policy)
+	}
+
+	return checkPlacementPolicyNotInUseFromMeta(t, policy)
+}
+
+func checkPlacementPolicyNotInUseFromInfoSchema(is infoschema.InfoSchema, policy *model.PolicyInfo) error {
+	for _, dbInfo := range is.AllSchemas() {
+		// TODO: check policy is not in use for databases
+		for _, tbl := range is.SchemaTables(dbInfo.Name) {
+			tblInfo := tbl.Meta()
+			if ref := tblInfo.PlacementPolicyRef; ref != nil && ref.ID == policy.ID {
+				return ErrPlacementPolicyInUse.GenWithStackByArgs(policy.Name)
+			}
+			// TODO: check policy is not in use for partitions
+		}
+	}
+	return nil
+}
+
+func checkPlacementPolicyNotInUseFromMeta(t *meta.Meta, policy *model.PolicyInfo) error {
```

Comment:
I added a dependency cache in `is`, pls help me with viewing this
https://github.com/pingcap/tidb/pull/28037

travel list may be costy

Reply:
I think I should close this pr because it seems #28937 has done all works for it.